In [1]:
import pandas as pd

Downloaded rawdata from https://www.unitedstateszipcodes.org/zip-code-database/

In [3]:
zip = pd.read_csv('raw_data/zip_code_database.csv')
zip.head()

zip      type  decommissioned primary_city acceptable_cities  \
0  501    UNIQUE               0   Holtsville               NaN   
1  544    UNIQUE               0   Holtsville               NaN   
2  601  STANDARD               0     Adjuntas               NaN   
3  602  STANDARD               0       Aguada               NaN   
4  603  STANDARD               0    Aguadilla             Ramey   

                                 unacceptable_cities state  \
0                           Internal Revenue Service    NY   
1                           Internal Revenue Service    NY   
2  Colinas Del Gigante, Jard De Adjuntas, Urb San...    PR   
3  Alts De Aguada, Bo Guaniquilla, Comunidad Las ...    PR   
4  Bda Caban, Bda Esteves, Bo Borinquen, Bo Ceiba...    PR   

                county             timezone area_codes world_region country  \
0       Suffolk County     America/New_York        631          NaN      US   
1       Suffolk County     America/New_York        631          NaN      US   
2   Adjuntas Municipio  America/Puerto_Rico    787,939          NaN      US   
3     Aguada Municipio  America/Puerto_Rico    787,939          NaN      US   
4  Aguadilla Municipio  America/Puerto_Rico        787          NaN      US   

   latitude  longitude  irs_estimated_population  
0     40.81     -73.04                       562  
1     40.81     -73.04                         0  
2     18.16     -66.72                         0  
3     18.38     -67.18                         0  
4     18.43     -67.15                         0

In [7]:
zip.country.value_counts()

US    42288
DE       84
JP       38
IQ       35
KR       26
      ...  
GI        1
GL        1
FI        1
RU        1
AQ        1
Name: country, Length: 61, dtype: int64

ZIP Codes range from 00501, belonging to the Internal Revenue Service in Holtsville, NY, to 99950 in Ketchikan, AK.

In [8]:
zip = zip[zip['country'] == 'US']
zip.shape

(42288, 15)

In [20]:
zip.isna().sum()

zip                             0
type                            0
decommissioned                  0
primary_city                    0
acceptable_cities           32986
unacceptable_cities         30615
state                           0
county                        489
timezone                      362
area_codes                   2590
world_region                42288
country                         0
latitude                        0
longitude                       0
irs_estimated_population        0
dtype: int64

In [29]:
zip = zip[['zip', 'primary_city', 'state', 'country']]

In [30]:
zip.isna().sum()

zip             0
primary_city    0
state           0
country         0
dtype: int64

In [33]:
zip['zipcode'] = zip['zip'].astype(str)

In [38]:
zip[zip['zipcode'].str.contains('^\d{,4}$', regex=True)]

zip primary_city state country zipcode
0      501   Holtsville    NY      US     501
1      544   Holtsville    NY      US     544
2      601     Adjuntas    PR      US     601
3      602       Aguada    PR      US     602
4      603    Aguadilla    PR      US     603
...    ...          ...   ...     ...     ...
3743  9880          Dpo    AE      US    9880
3744  9888          Apo    AE      US    9888
3745  9890          Apo    AE      US    9890
3746  9892          Dpo    AE      US    9892
3747  9895          Dpo    AE      US    9895

[3430 rows x 5 columns]

In [41]:
zip['zipcode'] = zip['zipcode'].str.zfill(5)
zip['zipcode']

0        00501
1        00544
2        00601
3        00602
4        00603
         ...  
42730    99926
42731    99927
42732    99928
42733    99929
42734    99950
Name: zipcode, Length: 42288, dtype: object

In [48]:
zip = zip.rename(columns = {'primary_city': 'city'})
zip = zip[['zipcode', 'city', 'state', 'country']]
zip.head()

zipcode        city state country
0   00501  Holtsville    NY      US
1   00544  Holtsville    NY      US
2   00601    Adjuntas    PR      US
3   00602      Aguada    PR      US
4   00603   Aguadilla    PR      US

In [49]:
zip.to_csv('preprocessed_data/zip_code.csv')